acnportal/acnsim/base.py

```diff
@@ -0,0 +1,649 @@
+"""
+This module contains a base class shared by all ACN-Sim objects.
+"""
+import json
+import operator
+import os
+# noinspection PyProtectedMember
+from pydoc import locate
+import warnings
+import pkg_resources
+import pandas
+from pandas.io.common import stringify_path, get_handle, get_filepath_or_buffer
+import numpy
+
+__NOT_SERIALIZED_FLAG__ = '__NOT_SERIALIZED__'
+
+
+# Uses methodology discussed in https://stackoverflow.com/a/16372436
+# by Martijn Pieters.
+
+
+def _operator_error_hooks(cls):
+    operator_hooks = [name for name in dir(operator)
+                      if name.startswith('__') and name.endswith('__')]
+
+    def add_hook(name):
+        # noinspection PyUnusedLocal
+        def op_hook(*args, **kwargs):
+            raise TypeError(
+                f"This object is a stub object whose methods are not "
```

Comment:
This is essentially the stack overflow answer you pointed me to with a custom error message. @zach401, does this message look okay?